sdks/python/apache_beam/typehints/typecheck.py

```diff
@@ -49,6 +49,13 @@ def __init__(self, dofn):
     super().__init__()
     self.dofn = dofn
 
+  def __getattribute__(self, name):
```

Comment:
also we should now be able to drop `setup` and `teardown` overrides.

Reply:
With this change the setup and teardown are no longer called in here:

https://github.com/apache/beam/blob/81cd93a8640c4f3994c68e091044e8e4e7c3f081/sdks/python/apache_beam/typehints/typecheck_test.py#L101-L122

Reverted for now: https://github.com/apache/beam/pull/27646/commits/81cd93a8640c4f3994c68e091044e8e4e7c3f081

Any way we can proceed as is? Otherwise please suggest where to start digging, thanks!